### PR TITLE
Add validation rules for aspect-ratio support via SSR

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #30291.

### CODE REVIEW

**Review by honeybadgerdontcare — APPROVED** [review]

**Comment by honeybadgerdontcare** [comment]
@dvoytenko `aspect-ratio` isn't in the allowed list which is why it failed. we're actually moving away from having an allowed list of declarations so that inline style will match the style tag rules but it hasn't been finished. In the mean time you can add `aspect-ratio` to the list here: https://github.com/ampproject/amphtml/blob/master/validator/validator-css.protoascii#L17

**Comment by Gregable** [comment]
If you want to update that list, ping wg-amp4email too to make sure that this should get allowed in email clients, or add it to the AMP_ONLY list.

**Comment by bill97819** [comment]
Adding `aspect-ratio` for email clients SGTM

File: validator/validator-css.protoascii [hunk 1]
```diff
@@ -29,6 +29,7 @@ declaration_list {
   declaration: { name: "animation-name" }
   declaration: { name: "animation-play-state" }
   declaration: { name: "animation-timing-function" }
+  declaration: { name: "aspect-ratio" }
```
Comment (zhangsu) [inline validator/validator-css.protoascii:32]: For the email spec, this only modifies the non strict mode spec. To allow it for strict mode spec, we need to add this to `EMAIL_SPECIFIC_DECLARATIONS`. The strict mode spec doesn't use `BASIC_DECLARATIONS`.

Comment (dvoytenko) [reply validator/validator-css.protoascii:32]: I added `aspect-ratio` to `EMAIL_SPECIFIC_DECLARATIONS` as well.

**Review by zhangsu — APPROVED** [review]

**Pull request merged**